lib/elixir/lib/exception.ex

```diff
@@ -1576,6 +1576,11 @@ defmodule UndefinedFunctionError do
     {"function #{formatted_fun} is undefined or private", true}
   end
 
+  defp message(:"undefined local", nil, function, arity) do
+    "nil." <> formatted_fun = Exception.format_mfa(nil, function, arity)
+    {"function #{formatted_fun} is undefined", false}
```

Comment:
Should we align this with IEx/compiler? It says:

    undefined function foo/0 (there is no such import)

Reply:
Sounds good.
Also, I should probably try to remove the hint as well, right now it is the default when `module == nil`:

```elxiir
iex(1)> defmodule X, do: (def env, do: __ENV__); Code.eval_quoted_with_env(quote(do: y()), [], X.env())
** (UndefinedFunctionError) function y/0 is undefined. If you are using the dot syntax, such as module.function(), make sure the left-hand side of the dot is a module atom
```

Reply:
[09437f1](https://github.com/elixir-lang/elixir/pull/12998/commits/09437f1c99a23f22472f8b4d4cdae2f6dce19385) and [a141b8e](https://github.com/elixir-lang/elixir/pull/12998/commits/a141b8ef81563b518abb5dc02f0170f2027b9219)